app/parsers/date_hash_parser.rb

```diff
@@ -21,7 +21,6 @@ def initialize(date_hash)
     @date_hash = date_hash
       .slice(:day, :month, :year)
       .compact_blank
```

Comment:
I wonder if things could be a little cleaner dealt with in this method chain, with something like:

```
    @date_hash = date_hash
      .slice(:day, :month, :year)
      .filter_map { |value| value.to_i if value.respond_to?(:to_i) && value.to_i.positive? }
      .compact
 ```
 
 Which would also get rid of needing the `non_positive_values?` method

Reply:
Agree, although there's a lot of logic in the `filter_map` which makes it a bit hard to grasp at a glance. And I think `filter_map` returns an Array which makes it a bit annoying as we'd still need to convert back!

How about using [`Kernel#Integer`](https://ruby-doc.org/3.4.1/Kernel.html#method-i-Integer) to avoid the quack check 🦆, followed by selecting present and positive values?:

```ruby
@date_hash = date_hash
  .slice(:day, :month, :year)
  .transform_values { |value| Integer(value, exception: false) }
  .select { |_, value| value&.positive? }
```

Then we also can do without the `#any_non_positive_values?` and `#to_i` calls on values below.

Reply:
Yeah that sounds better than my suggestion (good catch on filter_map) 👍 I didn't know about `exception: false` on Integer - that looks handy.